ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -5,102 +5,156 @@
 
 from __future__ import annotations
 
+import struct
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Literal, Optional, Union
+from enum import Enum
+from typing import Any, Callable, Optional
 
 from opentrons.hardware_control.types import OT3Axis
 
 from ot3_state_manager.ot3_state import OT3State
-from ot3_state_manager.util import Direction, get_md5_hash
+from ot3_state_manager.util import Direction, MoveMessageHardware, SyncPinState
+
+MESSAGE_BYTE_LENGTH = 4
+MESSAGE_ID_BYTE_LENGTH = 1
+MESSAGE_CONTENT_BYTE_LENGTH = 3
+STRUCT_FORMAT_STRING = f"B{MESSAGE_CONTENT_BYTE_LENGTH}s"
+
+
+# mypy doesn't like that I am instantiating a dataclass based on an abstract class (ABC)
```

Comment:
why not? is it a bug in mypy, or is there a good reason for it?

Reply:
It's based off of this [issue](https://github.com/python/mypy/issues/5374). Some people think it is a bug some don't. 

Based off of [this comment](https://github.com/python/mypy/issues/5374#issuecomment-568335302) I can turn the dataclass into a mixin and add it to a regular ABC. But it seemed messy.

Reply:
I ended up going a totally different route. 
`Message` does not need to be a dataclass. It doesn't have any data in it. 

Just making it an AbstractBaseClass that the concrete Message classes inherit from. 

5e9d5bd084a9179e33239417c6fa40880c955a57